lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -219,34 +219,48 @@ void StagesSubject::stageChanged(UsdNotice::ObjectsChanged const& notice, UsdSta
 			if (!sceneItem)
 				continue;
 
-			if (prim.IsActive())
+			// Special case when we know the operation came from either
+			// the add or delete of our UFE/USD implementation.
+			if (InAddOrDeleteOperation::inAddOrDeleteOperation())
 			{
-				if (InAddOrRemoveReference::inAddOrRemoveReference())
+				if (prim.IsActive())
 				{
-#if UFE_PREVIEW_VERSION_NUM >= 2014
-					// When we are in an add or remove reference we send the
-					// UFE subtree invalidate notif instead.
-					auto notification = Ufe::SubtreeInvalidate(sceneItem);
-					Ufe::Scene::notifySubtreeInvalidate(notification);
-#endif
+					auto notification = Ufe::ObjectAdd(sceneItem);
+					Ufe::Scene::notifyObjectAdd(notification);
 				}
 				else
 				{
-					auto notification = Ufe::ObjectAdd(sceneItem);
-					Ufe::Scene::notifyObjectAdd(notification);
+					auto notification = Ufe::ObjectPostDelete(sceneItem);
+					Ufe::Scene::notifyObjectDelete(notification);
 				}
 			}
 			else
 			{
-				auto notification = Ufe::ObjectPostDelete(sceneItem);
-				Ufe::Scene::notifyObjectDelete(notification);
+#if UFE_PREVIEW_VERSION_NUM >= 2014
+				// According to USD docs for GetResyncedPaths():
+				// - Resyncs imply entire subtree invalidation of all descendant prims and properties.
+				// So we send the UFE subtree invalidate notif.
+				auto notification = Ufe::SubtreeInvalidate(sceneItem);
+				Ufe::Scene::notifySubtreeInvalidate(notification);
+#endif
```

Comment:
Otherwise send the UFE subtree invalidate. Question: do we need anything for UFE < 0.2.14?

Reply:
No.  We only support the current Maya preview release, and it has 0.2.20.  We are supposed to go back and clean out these conditionals, but we rarely have.